src/components/Web3Status/ConnectWalletPopover.tsx

```diff
@@ -83,14 +93,23 @@ interface ConnectWalletProps {
 }
 
 export const ConnectWalletPopover = ({ setModal, tryActivation, children }: ConnectWalletProps) => {
-  const { connector } = useWeb3React()
+  const { connector, active, deactivate } = useWeb3React()
   const popoverRef = useRef<HTMLDivElement | null>(null)
   const walletSwitcherPopoverOpen = useModalOpen(ApplicationModal.WALLET_SWITCHER)
   const closeModals = useCloseModals()
   useOnClickOutside(popoverRef, () => {
     if (walletSwitcherPopoverOpen) closeModals()
   })
 
+  async function disconnect() {
+    try {
+      deactivate()
+      // Maybe inform user wallect has successfully disconnect?
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
```

Comment:
One more thing does it ever throw an error?

Reply:
that was mostly for us on the dev side. I'll remove it.